docs/reference/connectors/mongodb/extended-json.mdx

```diff
@@ -0,0 +1,52 @@
+---
+sidebar_position: 3
+sidebar_label: Extended JSON
+description:
+  "Types used by the MongoDB connector, and how those types map to GraphQL"
+keywords:
+  - mongodb
+  - configuration
+---
+
+# Extended JSON
+
+The MongoDB connector uses introspection to infer schemas for database collections. Those schemas specify a type for
+each document field. (See [Types](./connector-types.mdx)). Because MongoDB itself is schema-less there are cases where
+document data does not fit neatly into a type description. For these cases the special scalar type, `extendedJSON`, is
+provided as an escape hatch. `extendedJSON` is the connector's "any" type - a field or input of type `extendedJSON` can
+have any value.
+
+The situation that comes up most often is heterogeneous data: different documents in a collection contain different
+types of data under the same field name. For example consider a collection that has documents that have a `createdAt`
+field, and some of those documents use `date` values for that field while others use `string` values. Since Hasura does
+not implement union types there is no way to configure the connector for a field with type "either `date` or `string`".
+In this case the solution is to set the type of the field to `extendedJSON`. And that is what the introspection system
+does automatically when it samples data in this situation.
```

Comment:
This is great documentation, @hallettj:

- It clearly articulates why this type is present
- It walks through what's happening under the hood
- It makes it clear to a user when to reach for this and why

❤️